### ISSUE TITLE
Log of connection attempts

### ISSUE THREAD

**Issue by jmontoyaa** [chamilo/chamilo-lms#3810]
See BT#18167

**Comment by christianbeeznest** [comment]
Added in this PR https://github.com/chamilo/chamilo-lms/pull/4344

**Comment by christianbeeznest** [comment]
Added this request in this PR

**Comment by ywarnier** [comment]
I don't know why, but in case of successful login attempt by an admin, two logs are registered in track_e_login_record, at 2 seconds interval.

**Comment by ywarnier** [comment]
En base a BT#18167, hay unas cosas más que hacer:
- de la misma manera que se registran los logs en track_e_login_record, hay que escribir en un archivo de log en var/log/ids/ids.log
- el formato de log debería ser así: [{fecha}] [info] [client {dirección ip}] Login {succeeded/failed} for username {username}
- el archivo de logs debe tener una rotación diaria. Esto se puede obtener con un código como este:
```
$date = date('Y-m-d');
$dateTime = date('Y-m-d h:i:s');
$logFile = api_get_path(SYS_LOG_PATH).'ids/ids.log';
if (is_file($logFile)) {
    // read last line of file (see https://stackoverflow.com/a/1510248/6499848) and check the date
    if ($lastDate != $date) {
        // only if the last date is from yesterday or before that....
        // call a function to "+1" all log files
        // something like ($list = scandir(api_get_path(SYS_LOG_PATH).'ids/'); foreach ($list as $entry) { //avoid "." and ".." and files without a final number, then get the last bit of the file name and move it to $lastBit+1 }
        mv($logFile, $logFile.'.1');
    }
}
$logLine = '['.$dateTime.'] '.$message;
file_put_contents($logFile, $logLine, FILE_APPEND|LOCK_EX);
```

**Comment by ywarnier** [comment]
With the latest additions, this is *almost perfect*.
- one (and only one) record for each success or failure
- the file var/ogs/ids/ids-2024-03-01.log is created with the logs
- if I have logs from a previous day, then a new logfile is created to log the new attempts

However, the idea of naming the files with the date is not the same as naming the file "ids.log" and then rotating it to "ids.log.1" etc. The issue with that is that it doesn't use the same logic as normal log files in Linux, which are independent from a date because the frequency of rotation can be modified.
For example, you could say (later, in Chamilo) that you want one file per month (not one file per day) because it matches other logging processes for which you need one file per month.

So in your change here https://github.com/christianbeeznest/chamilo-lms/commit/6fcd5e909c15cc010520a28414de897e6b2bd839#diff-2245a80ab1d50200b20931dc2ff8e9366a2d09dda6f58da065a073c186c88278R34 the $logFilePrefix should not contain a date.

**Comment by ywarnier** [comment]
Fixed in 21ed7a7
